roles/jd-server/src/lib/job_declarator/mod.rs

```
@@ -90,113 +100,94 @@ impl JobDeclaratorDownstream {
-                    Err(e) => return Err(Box::new(JdsError::PoisonLock(e.to_string()))),
-                };
-                let tx = tx_.ok_or(JdsError::ImpossibleToReconstructBlock(
-                    "Missing transactions".to_string(),
-                ));
+            if let TransactionState::PresentInMempool(txid) = tx_with_state.1 {
+                let tx = mempool
+                    .safe_lock(|x| x.mempool.get(txid).cloned())
+                    .map_err(|e| JdsError::PoisonLock(e.to_string()))?
+                    .ok_or(JdsError::ImpossibleToReconstructBlock(
+                        "Txid not found in jds mempool".to_string(),
+                    ));
                 if let Ok(Some(tx)) = tx {
                     transactions_list.push(tx);
                 } else {
                     return Err(Box::new(JdsError::ImpossibleToReconstructBlock(
-                        "Missing transactions".to_string(),
+                        "Txid found in jds mempool but transactions not present".to_string(),
                     )));
                 }
             } else {
                 return Err(Box::new(JdsError::ImpossibleToReconstructBlock(
-                    "Missing transactions".to_string(),
+                    "Unknown transaction".to_string(),
                 )));
             };
         }
-        let block: Block =
-            roles_logic_sv2::utils::BlockCreator::new(last_declare, transactions_list, message)
-                .into();
-        Ok(hex::encode(serialize(&block)))
+        Ok(transactions_list)
+    }
+
+    async fn send_txs_to_mempool(self_mutex: Arc<Mutex<Self>>) {
+        let add_txs_to_mempool = self_mutex
+            .safe_lock(|a| a.add_txs_to_mempool.clone())
+            .unwrap();
+        let sender_add_txs_to_mempool = add_txs_to_mempool.sender_add_txs_to_mempool;
+        let add_txs_to_mempool_inner = add_txs_to_mempool.add_txs_to_mempool_inner;
+        let _ = sender_add_txs_to_mempool
```

why you ignore the error here?